phoenix-core-client/src/main/java/org/apache/phoenix/filter/PagingFilter.java

```diff
@@ -71,30 +68,21 @@ public void setDelegateFilter (Filter delegate) {
         this.delegate = delegate;
     }
 
-    public byte[] getRowKeyAtStop() {
+    public byte[] getCurrentRowKeyToBeExcluded() {
         byte[] rowKeyAtStop = null;
-        if (nextHintCell != null) {
-            // if we have already seeked to the next cell use that when we resume the scan
-            rowKeyAtStop = CellUtil.cloneRow(nextHintCell);
-        } else if (currentCell != null) {
+        if (currentCell != null) {
```

Comment:
Is it possible that currentCell is null? If so then PagingRegionScanner should check if the row key is null, and handle the null row key.

Reply:
currentCell would never be null because this is called only when paging filter stops. PagingFilter is stopped by:
```
        if (state == State.STARTED && currentCell != null
                && currentTime - startTime >= pageSizeMs) {
            state = State.TIME_TO_STOP;
        }
```

Hence, only if currentCell is not null, `TIME_TO_STOP` is set to stop the filter.